### ISSUE TITLE
Pre-trained Model

### ISSUE THREAD

**Issue by NikonD850** [Tyreal-Izual/DeTurb#1]
Greetings! I'm wonder that is it correct to use the models in /code/data named TMT_x.pth for recurrence your experiment? The filenames make me confused since your repository is called DeTurb.

**Comment by Tyreal-Izual** [comment]
Hi! Thanks for reaching out.

Yes, you're correct. The models in `/code/data` named `TMT_x.pth` are indeed used to reproduce the experiments. Apologies for the confusing naming. This repository was originally created for my undergraduate thesis project, and some of the earlier filenames (like TMT) carried over from earlier versions.

Since I’ve been away from the university research lab for a year during my postgraduate studies, the repository hasn’t been actively maintained. I’m planning to clean it up and make everything more readable and consistent in the near future.

Thanks for your patience, and feel free to reach out if you have any other questions!

**Comment by Tyreal-Izual** [comment]
Hi! Thank you for your interest in Deturb and for the effort you have already invested in retraining the model.

Unfortunately, I’m afraid I no longer have a standalone checkpoint for the single stage network: the temporary Stage-1 weights we used during that ablation were stored on a decommissioned account, so I no longer have access to them. 

A quick clarification: Nearly every experiment reported in our paper including the quantitative tables was run with the full-stage model, the single-stage variant appears only in the ablation study, which you may not achieve the same result using single model. 

And at the moment I do not have plans or the computational resources to retrain that single module from scratch.

Given this, I would strongly recommend using or retraining the full two-stage model. I've also update the README for better look. 

If you can post your training log / val curves or sample outputs, I’m happy to take a look.

Sorry I can’t provide the exact file you were after, but I hope this clarifies the situation. Feel free to reach out with any further questions.

**Comment by LCH0528** [comment]
OK! In the zip is the result of the dynamic part of the TMT dataset that I used to train with (untrained static). The first folder in it uses the hyperparameters for batch_size=1; the second is for batch_size=4, which is relatively complete. However, from the RECORDING file, it can be seen that no matter which one it is, the parameters such as LOSS, which have been validated after every 5 rounds of training, have not changed, and I wonder if there is a problem in any part of the process. You It is recommended to use the one in the second folder as the main one.


In addition, the rest of the .py files in the attachment sent are the corresponding model and training files. The Among them, the core of the model file has not been changed, but a main has been added; the training file has changed the training process to mixed precision training.
That's all that's currently in the section about the first stage of geometric correction that contains deformable convolution, and the folder of results from the second stage of training with it after training is not currently in my possession (in the lab); however, I've looked at it, and it doesn't look as good as the training results for the original plain UNet3d in the TMT paper, so I'll just post what I've just described for the time being.


If you want anything else relevant, please contact me; and if there are any suggestions for what has just been described, please contact me as well. I am actually a graduate student in an ordinary school, there is no atmospheric turbulence video de-blurring around the neighborhood, relatively cold (more target detection, segmentation, etc.), and not much experience, no one to collaborate on guidance and so on, what is my humble opinion 
Please also point out.


If you are in a position to do so, you may consider sending it to your colleagues, friends, etc. for assistance, analysis and evaluation.


Thanks for the previous reply!




译文：


好的！压缩包中是我曾经用TMT数据集的dynamic部分训练的结果（未训练static）。其中，里面的第一个文件夹用的是batch_size=1的超参数；第二个是batch_size=4的，相对来说比较完整。不过，从recording文件中可以看出，不管是哪一个，每5轮训练后经过验证的loss等参数都没变，本人怀疑是不是哪个环节有问题。建议在使用的时候以第二个文件夹的为主。


此外，发送的附件中剩下的.py文件是对应的model和训练文件。其中，model文件核心没改，就加了个main；训练文件是把训练过程改成了混合精度训练。
目前关于第一阶段几何纠正的部分中含有可变形卷积的就这么多，训练完后用其训练第二阶段的结果文件夹，目前不在我手上（在实验室里）；不过我看过了，没有TMT论文中原来的普通UNet3d的训练结果好，我就暂时发刚刚所述的东西了。


如果你还想要其他相关的，请联系我；如果对于刚刚所述的有什么建议，也请联系我。本人其实也是个普通学校的研究生，周围附近没有搞大气湍流视频去模糊的，相对来说比较冷门（较目标检测、分割等），也没什么经验，没人合作指导啥的，有什么拙见
也请指点。


刚刚发的东西，如果有条件的化，也可以考虑发给您的同事、朋友等人进行协助、分析和评判。


感谢之前的回复！





------------------ 原始邮件 ------------------
发件人:                                                                                                                        "Tyreal-Izual/DeTurb"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年7月13日(星期天) 凌晨2:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Tyreal-Izual/DeTurb] Pre-trained Model (Issue #1)



Tyreal-Izual left a comment (Tyreal-Izual/DeTurb#1)
 
Hi! Thank you for your interest in Deturb and for the effort you have already invested in retraining the model.
 
Unfortunately, I’m afraid I no longer have a standalone checkpoint for the single stage network: the temporary Stage-1 weights we used during that ablation were stored on a decommissioned account, so I no longer have access to them.
 
A quick clarification: Nearly every experiment reported in our paper including the quantitative tables was run with the full-stage model, the single-stage variant appears only in the ablation study, which you may not achieve the same result using single model.
 
And at the moment I do not have plans or the computational resources to retrain that single module from scratch.
 
Given this, I would strongly recommend using or retraining the full two-stage model. I've also update the README for better look.
 
If you can post your training log / val curves or sample outputs, I’m happy to take a look.
 
Sorry I can’t provide the exact file you were after, but I hope this clarifies the situation. Feel free to reach out with any further questions.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;


从QQ邮箱发来的超大附件

UNet3dDC.rar (3.27G, 2025年10月11日 08:06 到期)进入下载页面：https://mail.qq.com/cgi-bin/ftnExs_download?k=ceca64634d0a74e7fcb91763343565357e8d5ab236356535141a411703575c540251145a06565c18060c0d571b565c5354180902535454560103085354056935647b5c17055121761f47581122a4672d9f6efc63b4800bbb91dba896b95d409b67&t=exs_ftn_download&code=159c65e5

**Comment by Tyreal-Izual** [comment]
HI, thanks for sharing the information. 

I couldn't see any .py file in your shared file, but based on the folder name and training log file, it seems like you were training based on [deformConv_train_tilt_dynamic.py](https://github.com/Tyreal-Izual/DeTurb/blob/main/code/deformConv_train_tilt_dynamic.py), which is an outdated version and we don't use that anymore. 

If so, please try to train the model using the `train_TMT_2stage.py` file, and follow the Training Instructions in README, thanks. 

I'm currently involved in a heavy program and don't have time to refactor the file names for this project in a clear way. Sorry for any inconvenience that this could cause. I will update them in the near future.
